Assets/SEE/Scanner/TokenMetrics.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace SEE.Scanner
+{
+    /// <summary>
+    /// Provides metrics calculated based on code tokens.
+    /// </summary>
+    public static class TokenMetrics
+    {
+        /// <summary>
+        /// Calculates the McCabe cyclomatic complexity for provided code.
+        /// </summary>
+        /// <param name="tokens">The tokens used for which the complexity should be calculated.</param>
+        /// <returns>Returns the McCabe cyclomatic complexity.</returns>
+        public static int CalculateMcCabeComplexity(IEnumerable<SEEToken> tokens)
+        {
+            int complexity = 1; // Starting complexity for a single method or function.
+
+            // Count decision points (if, for, while, case, &&, ||, ?).
+            complexity += tokens.Count(t => t.TokenType == SEEToken.Type.Keyword);
```

Comment:
This is not the McCabe complexity. You are just counting all keywords. You need to count only the keywords that indicate branches (if, while. for, case, etc.).

The `SEEToken` is actually not the right data structure to make this decision. `SEEToken` is an abstraction of  tokens to determine the syntax highlighting. At the level of `SEEToken`, the distinction of the kind of tokens (if, while, class, etc.) is lost. You need to gather the metrics at the raw tokens provided by the underlying scanners generated by AntLR.